contracts/mesh-provider/src/state.rs

```diff
@@ -51,6 +61,43 @@ impl Stake {
         val.shares_to_tokens(self.shares)
     }
 
+    /// Calculate rewards
+    pub fn calc_pending_rewards(
+        &mut self,
+        new_rptpb: Uint128,
+        staked: Uint128,
+        height: u64
+    ) -> Result<(), ContractError> {
+        if staked.is_zero() || self.rewards.last_height == 0 {
+            self.rewards.last_rptpb = new_rptpb;
+            self.rewards.last_height = height;
+            return Ok(());
+        }
+
+        let rptpb_to_pay = new_rptpb - self.rewards.last_rptpb;
+        let block_to_pay = height - self.rewards.last_height;
+
+        if rptpb_to_pay.is_zero() || block_to_pay == 0 {
+            self.rewards.last_rptpb = new_rptpb;
+            self.rewards.last_height = height;
+            return Err(ContractError::ZeroRewardsToSend {})
```

Comment:
Is it good it errors here?
If it is called to "pay out pending" many other places, best is to return `Ok(())` for "do nothing"

Also, why update the variables if we then error (reverting the tx)?

Reply:
My guess is 'late night coding', will look into it again.